eval/eval_dd_package.cpp

```diff
@@ -161,22 +161,22 @@ benchmarkFunctionalityConstructionGrover(
   const std::bitset<128U> iterBits(iterations);
   const auto msb = static_cast<std::size_t>(std::floor(std::log2(iterations)));
   auto f = iter;
-  dd.incRef(f);
+  dd.track(f);
   bool zero = !iterBits[0U];
   for (std::size_t j = 1U; j <= msb; ++j) {
     f = dd.applyOperation(f, f);
     if (iterBits[j]) {
       if (zero) {
-        dd.incRef(f);
-        dd.decRef(e);
+        dd.track(f);
+        dd.untrack(e);
         e = f;
         zero = false;
       } else {
-        e = dd.applyOperation(e, f);
+        e = dd.applyOperation(f, e);
```

Comment:
Just to point out the obvious (or maybe not): this line here is responsible for messing with the statistics for the Grover functionality construction.
Given how it's actually a bugfix though, these can be safely ignored.
It should be more correct now.
In the previous case, it could happen that numbers with a zero reference count got decremented once more, which effectively immortalised them through an underflow of the ref count.

Reply:
Thanks for running these again! 👍🏼 
I think it really boils down to the difference in garbage collection runs. If an additional run triggers, this will considerably increase the runtime due to the complexity of the collection. This phenomenon is independent of the PR changes here though and it might as well happen if you run the benchmarks multiple times on the main branch.

Given how it seems to be a 5-15% runtime improvement for most of the remaining benchmarks, I would consider this a success overall 💪🏼